### PR TITLE
chore(service naming): add _dd.base_service tag

### PR DESCRIPTION
**Pull request by tabgok**
This change is being done to propagate global service information to the backend for the purpose of improving the Service Map.  It has no affect on the tracer beyond adding additional information.

Prior to this change spans did not contain any contextual information about the global configuration if span.service was set.

After this change, all spans whose service differs from the globally configured value will contain a `_dd.base_service` tag which contains the globally configured value.

Note that the global value by default is set to the value of the environment variable DD_SERVICE

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by jbertran — DISMISSED** [review]
Case insensivity should be added, and minor nits.

File: tests/contrib/django/test_django.py [hunk 1]
```diff
@@ -1423,13 +1423,15 @@ def test_cached_view(client, test_spans):
             "views.decorators.cache.cache_page..GET.03cdc1cc4aab71b038a6764e5fcabb82.d41d8cd98f00b204e9800998ecf8..."
         ),
         "_dd.p.dm": "-0",
+        "_dd.base_service": "",
```
[the remark this answers is not in the document]
Comment (tabgok) [reply tests/contrib/django/test_django.py:1426]: This is the latest decision, based upon a discussion with the squad

[the remark this answers is not in the document]
Comment (tabgok) [reply tests/contrib/django/test_django.py:1426]: verified that it is the correct decision

File: ddtrace/span.py [hunk 2]
```diff
@@ -254,13 +256,18 @@ def duration(self, value):
         # type: (float) -> None
         self.duration_ns = int(value * 1e9)
 
+    def _set_dd_base_service(self):
```
[the remark this answers is not in the document]
Comment (tabgok) [reply ddtrace/span.py:259]: I'm going to ask the python guild about that

File: ddtrace/span.py [hunk 3]
```diff
@@ -254,13 +256,18 @@ def duration(self, value):
...
+        if self.service not in (None, config.service):
```
[the remark this answers is not in the document]
Comment (tabgok) [reply ddtrace/span.py:260]: fixed

**Review by majorgreys — APPROVED** [review]

**Pull request merged**